blocks/search-results/search-results.js

```diff
@@ -2,15 +2,21 @@ class SearchResults {
   constructor(element, options) {
     this.element = element;
     this.options = options;
+
+    window.addEventListener('cn:search', () => this.init());
+    window.addEventListener('popstate', () => this.init());
   }
 
   init() {
-    this.searchterm = SearchResults.getSearchTerm();
+    const newSearchTerm = SearchResults.getSearchTerm();
+    if (this.searchterm !== newSearchTerm) {
```

Comment:
it is preferable to remove the typed text than not let the user reuse the search, because if a change is made that affect the recent results, then the search doesn't reflect the newly changed data

Reply:
I prefer to leave it re-searchable without having to type it again.

Reply:
what sense does it make to do the exact search again? would we get the same result anyway?

Reply:
not necessarily. Let's do it the other way. What sense does it make to hack the search solution in the frontend?

Reply:
what part of this is a hack? It just avoids unnecessary requests to the backend and unwanted searches because also the browser history gets polluted with the same search multiple times, but if you all think it is a valid use case to search for the same thing multiple times and expect different results 🤷 I'll remove it and add a check to at least not have the same history entry multiple times.

Reply:
There is no issue with having the same history multiple times either.
Please, let's not hack the interaction between the user and search.

Reply:
Again, I am not hacking anything, setting the browser history we need to do as we are staying on the same page for multiple requests , SPA style, otherwise the user cannot navigate as expected, having the exact same entry multiple times in a row in the history makes no sense.

Reply:
If the user decided to perform those actions during the navigation, it makes sense for the user, and therefore for us.

Reply:
I think this discussion goes beyond code quality we should discuss in a PR and should rather be discussed/decided by an UX expert. As the code introduced with this PR works and there seems to be no further concerns about the code quality I propose to merge it and open a separate ticket if the UX should be further improved.

Reply:
I don't get your intentions.

Reply:
This if is critical for the search and you are adding it with no need. You have a search specialist here. Check with UX, this is fine, but do not add an if that the search specialist didn't ask you for and it is telling you that will damage the search solution.

Reply:
This is not damaging anything and there is generally a need to add logic to make the browser history still work when staying on the same page the question we are discussing here is how this custom logic should look like and I think the version we have right now works quite good.